crates/typst-library/src/layout/container.rs

```diff
@@ -338,23 +359,29 @@ pub struct BlockElem {
     #[default(false)]
     pub clip: bool,
 
-    /// Whether this block must stick to the following one, with no break in
-    /// between.
+    /// If and how this block must stick to the blocks surrounding it, with no
+    /// break in between.
     ///
-    /// This is, by default, set on heading blocks to prevent orphaned headings
-    /// at the bottom of the page.
+    /// This is, by default, set to "below" on heading blocks to prevent
+    /// orphaned headings at the bottom of the page.
     ///
     /// ```example
     /// >>> #set page(height: 140pt)
     /// // Disable stickiness of headings.
-    /// #show heading: set block(sticky: false)
+    /// #show heading: set block(sticky: none)
     /// #lorem(20)
     ///
     /// = Chapter
     /// #lorem(10)
+    ///
+    /// #table(
+    /// columns: 2,
+    /// [A], [B],
+    /// [C], [D],
+    /// )
+    /// #block(sticky: "above")[The above table shows that...]
     /// ```
-    #[default(false)]
```

Comment:
Let's add some compatibility behavior to temporarily accept bools with a warning. (You can leave this for last, after all other reviews have been addressed.)